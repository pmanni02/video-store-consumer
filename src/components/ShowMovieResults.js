import React, { Component } from 'react';
import PropTypes from 'prop-types';
import TmbdMovie from './TmbdMovie';

class ShowMovieResults extends Component {
  constructor(props){
    super(props);
  }

  statusUpdate = (status) => {
    this.props.tmbdStatusCallback(status);
  }

  renderMovies = () => {
    const movieList = this.props.movies.map((movie, index) => {
      return(
        <TmbdMovie
          key = { index }
          id = { movie.external_id }
          title = { movie.title }
          releaseDate = { movie.release_date }
          poster = { movie.image_url}
          overview = { movie.overview }
          statusCallback = { this.statusUpdate }
        />
      )
    })

    return movieList;
  }


  render(){
    return(
      <div>
        { this.renderMovies() }
      </div>
    );
  }
}

ShowMovieResults.propTypes = {
  movies: PropTypes.array.isRequired,
  tmbdStatusCallback: PropTypes.func.isRequired
};


export default ShowMovieResults;
