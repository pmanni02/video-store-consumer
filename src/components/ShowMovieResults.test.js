import React from 'react';
import { shallow } from 'enzyme';
import ShowMovieResults from './ShowMovieResults';

describe('ShowMovieResults', () => {
  test('snapshot', () => {
    const wrapper = shallow(
      <ShowMovieResults
        movies={[]}
        tmbdStatusCallback={()=>{}}
      />
    );
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });
});
