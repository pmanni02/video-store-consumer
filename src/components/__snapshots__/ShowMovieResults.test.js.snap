// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowMovieResults snapshot 1`] = `
<div
  className="tmbd-movie-list"
/>
`;
